vamb/vambtools.py

```diff
@@ -677,7 +677,7 @@ def write_bins(
                 )
 
         # Print bin to file
-        with open(directory.joinpath(binname + ".fna"), "wb") as file:
+        with open(directory.joinpath(str(binname) + ".fna"), "wb") as file:
```

Comment:
Why is this `str` necessary? According to the function's type signature, it should already be a string. If it is always a string, do not use `str` here. If it is not always, then the type signature needs to be updated, or the caller needs to make sure the binname is always a string (in order to conform to the signature)